csv/src/main/java/org/apache/metamodel/csv/CsvConfiguration.java

```diff
@@ -195,4 +199,17 @@ public String toString() {
                 + ", separatorChar=" + separatorChar + ", quoteChar=" + quoteChar + ", escapeChar=" + escapeChar
                 + ", failOnInconsistentRowLength=" + failOnInconsistentRowLength + "]";
     }
+    
+    public ICSVParser createParser() {
+        if (getEscapeChar() == getQuoteChar()) {
+            return new RFC4180ParserBuilder().withSeparator(getSeparatorChar()).withQuoteChar(getQuoteChar()).build();
+        } else {
+            return new CSVParserBuilder()
+                    .withSeparator(getSeparatorChar())
+                    .withQuoteChar(getQuoteChar())
+                    .withEscapeChar(getEscapeChar())
+                    .build();
```

Comment:
Does the builder also have thread safety issues? Otherwise it might be a good idea to save instantiations by keeping the builder instead of using a new one every time.

Reply:
As far as I know the builder doesn't have those issues, but I refrained from that up until now because the _RFC4180ParserBuilder_ and the _CSVParserBuilder_ don't share one super class or interface.

Reply:
Ah, yes, that definitely makes it harder to do in a sane way (and is a bit silly, given that they at least share the `build()` method).